centreon/src/Core/HostTemplate/Application/UseCase/PartialUpdateHostTemplate/PartialUpdateHostTemplate.php

```diff
@@ -165,6 +179,199 @@ private function updatePropertiesInTransaction(PartialUpdateHostTemplateRequest
         }
     }
 
+    /**
+     * @param PartialUpdateHostTemplateRequest $request
+     * @param HostTemplate $hostTemplate
+     *
+     * @throws \Throwable|AssertionFailedException|HostTemplateException
+     */
+    private function updateHostTemplate(PartialUpdateHostTemplateRequest $request, HostTemplate $hostTemplate): void
+    {
+        $this->info('PartialUpdateHostTemplate: update host template', ['host_template_id' => $hostTemplate->getId()]);
+
+        $inheritanceMode = $this->optionService->findSelectedOptions(['inheritance_mode']);
+        $inheritanceMode = isset($inheritanceMode['inheritance_mode'])
+            ? (int) $inheritanceMode['inheritance_mode']->getValue()
+            : 0;
+
+        if (
+            ! $request->name instanceOf NoValue
+            && $hostTemplate->getName() !== HostTemplate::formatName($request->name)
+        ) {
+            $this->validation->assertIsValidName($request->name);
+            $hostTemplate->setName($request->name);
```

Comment:
I am annoyed by the `HostTemplate::formatName` call which is not obvious regarding the `->setName`.
I understand why this is done, but that's too far from "self-documented code" mindset.
I haven't a better suggestion right now, I must think about this and we need to discuss in meet.